CMSgov/design-system
# [WNMGDS-731] Update responsive stacked table

## Summary
Update the stacked table component to more closely align with the recently released [USWDS responsive stacked table design](https://designsystem.digital.gov/components/table/#responsive-stacked-table).

## Updated
- Layout of stacked table

## Added
- Stacked table with headers variant via `stackableStyles="header"` props

### Stacked table layout
#### Stacked bordered table - new layout
![Screen Shot 2021-01-27 at 2 10 14 PM](https://user-images.githubusercontent.com/20322880/106041192-7c889f00-60a9-11eb-86bd-55f92609d6c2.png)

#### Stacked table - old layout
![Screen Shot 2021-01-27 at 2 09 56 PM](https://user-images.githubusercontent.com/20322880/106041170-75fa2780-60a9-11eb-992d-57b3607ee57c.png)

### How to test
- Ensure stacked table works on various browsers and is accessible by screen readers
- http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-731/Update-responsive-stacked-table/
